From cb52a5f0ff5ed38562b94c3a084efac7c9d8161a Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 2 Dec 2016 14:40:45 -0800
Subject: [PATCH] RELENG-735 headnode builds sometimes fail with 'lofiadm:
 could not unmap device /dev/lofi/2: Device busy' Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0091ce2..c2a37a6 100644
--- a/Makefile
+++ b/Makefile
@@ -2532,7 +2532,12 @@ BOOT_OUTPUT=$(USB_BITS_DIR)/boot$(HEADNODE_SUFFIX)-$(_headnode_stamp).tgz
 .PHONY: cleanimgcruft
 cleanimgcruft:
 	$(RM) -vf /tmp/*4gb.img
-	for dev in $(shell lofiadm | cut -d ' ' -f1 | grep -v "^Block"); do pfexec lofiadm -d $${dev}; done
+	for dev in $(shell lofiadm | cut -d ' ' -f1 | grep -v "^Block"); do  \
+		mount | grep "on $${dev}" | cut -d' ' -f1 | while read mntpath; do \
+			umount $${mntpath}; \
+		done; \
+		pfexec lofiadm -d $${dev}; \
+	done
 
 .PHONY: boot
 boot: $(BOOT_OUTPUT)
-- 
2.21.0

